docs/registry/README.md

```diff
@@ -0,0 +1,193 @@
+# Registry Proposal
+
+## Objective
+
+Design an **initial** version of the Registry that can support discoverability of 
+the different event types that can be consumed from the eventing mesh.
+
+## Requirements
+
+Our design revolves around the following core requirements:
+
+1. We should have a Registry per namespace to enforce isolation.
+2. The Registry should contain the event types that can be consumed from
+the eventing mesh. If an event type is not ready for consumption, we should explicitly indicate so.
+3. The event types stored in the Registry should contain all the required information 
+for a consumer to create a Trigger without resorting to some other OOB mechanism.
+
+## Design Ideas
+
+### EventType CRD
+
+We propose having a namespaced EventType CRD. Here is an example of how a CR would look like:
+
+```yaml
+apiVersion: eventing.knative.dev/v1alpha1
+kind: EventType
+metadata:
+  name: repopush
+spec:
+  type: repo:push
+  source: my-user/my-repo
+  schema: my-schema
+  broker: default
+```
+
+
+- The `name` of the EventType is advisory, non-authoritative. Given that Cloud Event types can 
+contain characters that may not comply with Kubernetes naming conventions, we will (slightly) 
+modify those names to make them K8s-compliant, whenever we need to generate them. 
+
+- `type` is authoritative. This refers to the Cloud Event type as it enters into the eventing mesh. 
+
+- `source`: an identifier of where we receive the event from. This might not necessarily be the Cloud Event source 
+attribute. If we receive the event from our receive adaptors, the info might come in a Cloud Event custom extension (e.g., from).
+
+- `schema` is a URI with the EventType schema. It may be a JSON schema, a protobuf schema, etc. It is optional.
```

Comment:
Forgive me if my understanding is wrong. I think here `source` and `schema` describes the necessary parameters of an EventType. EventType to Event is as Class to Object. 

I think parameters to a specific kind of event source may be divided into below types:
- authentication
- uri to identify this event source, e.g. ip:address to a database server or a message broker, url for a GitHub repo, and etc
- others, e.g. topic in a Kafka event, namespace in a K8s event, and etc.

Where do you plan to put authentication values then ?

Reply:
`source` is mandatory, yes.
`schema` is optional. 
Those parameters are used to identify the EventType that can flow through the system.

Reply:
> I think parameters to a specific kind of event source

Each source has its way of specifying the authentication information. GitHubSource uses secrets, and so on. But that is for sources, not for EventTypes.

Reply:
Yes, you are right. 
The "source" field refers to the CloudEvents source.
Event Source is the "code that produces events", e.g., GitHubSource.